src/main/java/com/csfrequency/firebase/devicestore/FirebaseDeviceStore.java

```diff
@@ -142,70 +155,19 @@ public void unsubscribe() {
         subscribed = false;
     }
 
-    private void deleteDevice(final String userId) {
+    private Task<Void> deleteDevice(final String userId) {
         final DocumentReference docRef = userRef(userId);
-
-        firestore.runTransaction(new Transaction.Function<Void>() {
-            @Override
-            public Void apply(Transaction transaction) throws FirebaseFirestoreException {
-                DocumentSnapshot doc = transaction.get(docRef);
-
-                if (doc.exists()) {
-                    List<Map<String, String>> devices = getDevices(doc);
-                    // Remove the old device
-                    devices = removeCurrentDevice(devices);
-                    // Update the document
-                    transaction.update(docRef, DEVICES_FIELD, devices);
-                } else {
-                    Map<String, Object> userDevices = createUserDevices(userId, null);
-                    transaction.set(docRef, userDevices);
-                }
-                return null;
-            }
-        });
+        return docRef.update(FieldPath.of(DEVICES_FIELD, getDeviceId()), FieldValue.delete());
```

Comment:
One final note: This will now fail the deletion if the document doesn't exist. If you use `set()` with `SetOptions.merge()` the task will succeed even if there is no document (and create a document instead).

Reply:
There _should_ always be a document to update, as the library will have automatically created a device if they are logged in at application startup.

I can change if you think it's worth protecting against, but it does make the code less readable given the lack of a `map` shorthand in Java 8.

Reply:
I'm gonna leave this up to you. I do think the current code is a little cleaner.

Reply:
I'm going to leave as is for now - if it proves to be an issue, then it can be revisited